src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/GetAzureADApplicationCommand.cs

```diff
@@ -52,21 +52,23 @@ public override void ExecuteCmdlet()
             }
             else
             {
-                ApplicationFilterParameters parameters = new ApplicationFilterParameters();
+                Rest.Azure.OData.ODataQuery<Application> odataQueryFilter = null;
+
                 if (ApplicationId != Guid.Empty)
                 {
-                    parameters.AppId = ApplicationId;
+                    string appId = ApplicationId.ToString();
+                    odataQueryFilter = new Rest.Azure.OData.ODataQuery<Application>(a => a.AppId == appId);
                 }
                 else if (!string.IsNullOrEmpty(DisplayNameStartWith))
                 {
-                    parameters.DisplayNameStartsWith = DisplayNameStartWith;
+                    odataQueryFilter = new Rest.Azure.OData.ODataQuery<Application>(a => a.DisplayName.StartsWith(DisplayNameStartWith));
                 }
                 else if (!string.IsNullOrEmpty(IdentifierUri))
                 {
-                    parameters.IdentifierUri = IdentifierUri;
+                    odataQueryFilter = new Rest.Azure.OData.ODataQuery<Application>(a => a.IdentifierUris.Contains(IdentifierUri));
                 }
 
-                WriteObject(ActiveDirectoryClient.GetApplicationWithFilters(parameters), enumerateCollection: true);
+                WriteObject(ActiveDirectoryClient.GetApplicationWithFilters(odataQueryFilter), enumerateCollection: true);
```

Comment:
@shuagarw  Is there a change in the output of thic cmdlet?  If so, we will need to document the change in release notes.